src/Microsoft.Azure.SignalR/HubHost/NegotiateHandler.cs

```diff
@@ -77,13 +101,14 @@ public async Task<NegotiationResponse> Process(HttpContext context)
             }
 
             var queryString = GetQueryString(request.QueryString.HasValue ? request.QueryString.Value.Substring(1) : null, cultureName);
+            var availableTransports = GetAvailableTransports(_transportTypeProvider.Invoke(context));
 
             return new NegotiationResponse
             {
                 Url = provider.GetClientEndpoint(_hubName, originalPath, queryString),
                 AccessToken = await provider.GenerateClientAccessTokenAsync(_hubName, claims),
                 // Need to set this even though it's technically protocol violation https://github.com/aspnet/SignalR/issues/2133
-                AvailableTransports = new List<AvailableTransport>()
+                AvailableTransports = availableTransports
```

Comment:
client does not respect this `AvailableTransports` when it is a redirect response, we need to pass the info through claim to the service, and let service return `AvailableTransports`